pydantic/types.py

```diff
@@ -1008,3 +1012,34 @@ def to(self, unit: str) -> float:
             raise errors.InvalidByteSizeUnit(unit=unit)
 
         return self / unit_div
+
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ DATE TYPES ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+class PastDate(date):
```

Comment:
I reckon you should add
```py
if TYPE_CHECKING:
    PastDate = date
    FutureDate = date
```
and add two lines in `tests/mypy/modules/success.py` in `PydanticTypes` class

Reply:
Just a dumb model with most custom pydantic types to ensure `mypy` is happy 😊